eZ/Publish/Core/Search/Legacy/Content/WordIndexer/Gateway/DoctrineDatabase.php

```diff
@@ -0,0 +1,476 @@
+     */
+    public function bulkIndexContent(array $contentObjects)
+    {
+        foreach ($contentObjects as $contentObject) {
+            $this->index($contentObject);
+        }
+    }
+
+    /**
+     * Remove entire search index
+     */
+    public function purgeIndex()
+    {
+        $this->searchIndex->purge();
+    }
+
+    /**
+     * Index wordIndex
+     *
+     * Ported from the legacy code
+     * @see https://github.com/ezsystems/ezpublish-legacy/blob/master/kernel/search/plugins/ezsearchengine/ezsearchengine.php#L255
+     *
+     * @param Content $content
+     * @param $indexArray
+     * @param $wordIDArray
+     * @param int $placement
+     * @return int last placement
+     */
+    private function indexWords(Content $content, $indexArray, $wordIDArray, $placement = 0)
+    {
+        $contentId = $content->versionInfo->contentInfo->id;
+
+        // Count the total words in index text
+        // $totalWordCount = count($indexArray);
+
+        /*
+        // Needs to be rewritten
+        // Count the number of instances of each word
+        $wordCountArray = array_count_values($indexArray);
+        // Strip double words
+        $uniqueIndexArray = array_unique($indexArray);
+        // Count unique words
+        $uniqueWordCount = count($uniqueIndexArray);
+        */
+
+        $prevWordId = 0;
+        $contentTypeId = $content->versionInfo->contentInfo->contentTypeId;
+        $sectionId = $content->versionInfo->contentInfo->sectionId;
+        $published = $content->versionInfo->contentInfo->publicationDate;
+
+        for ($i = 0; $i < count($indexArray); $i++) {
+            $indexWord = $indexArray[$i]['Word'];
+            $contentFieldId = $indexArray[$i]['ContentClassAttributeID'];
+            $identifier = $indexArray[$i]['identifier'];
+            $integerValue = $indexArray[$i]['integer_value'];
+            $wordId = $wordIDArray[$indexWord];
+
+            if (isset($indexArray[$i + 1])) {
+                $nextIndexWord = $indexArray[$i + 1]['Word'];
+                $nextWordId = $wordIDArray[$nextIndexWord];
+            } else {
+                $nextWordId = 0;
+            }
+            // Calculate the relevance ranking for this word
+//            $frequency = ( $wordCountArray[$indexWord] / $totalWordCount );
+            $frequency = 0;
+            $this->searchIndex->addObjectWordLink($wordId, $contentId, $frequency, $placement, $nextWordId, $prevWordId, $contentTypeId, $contentFieldId, $published, $sectionId, $identifier, $integerValue);
+            $prevWordId = $wordId;
+            $placement++;
+        }
+
+        return $placement;
+    }
+
+    /**
+     * Get field meta data
+     *
+     * Based on the legacy code
+     *
+     * @todo move this elsewhere to get rid of switch statement
+     *
+     * @param Content\Field $field
+     * @return string
+     */
+    private function getMetaData(Content\Field $field)
```

Comment:
If the goal is to return a searchable value of a field, I would not call it getMetaData... :-)